### ISSUE TITLE
[Bug]: The wallpaper on homepage is removed after accessing the Settings menu

### ISSUE THREAD

**Issue by delia-pop** [mozilla-mobile/fenix#26999]
### Steps to reproduce

1. Have an old version with a wallpaper set. > Update the app to latest version.
2. OR fresh install the latest version > Set a wallpaper from the Wallpapers Settings menu.
3. Observe the homepage.
4. Access a webpage/access the Settings menu and return to homepage. Observe the wallpaper on homepage.
5. Restart the app and observe the homepage.

### Expected behaviour

The wallpaper set remains displayed on homepage when accessing different screens and returning to homepage.

### Actual behaviour

After dismissing the onboarding, the wallpaper set in Steps 1 or 2 is displayed on homepage. 
However:
- Opening a webpage and browsing does not remove the wallpaper;
- Restarting the app does not remove the wallpaper, but issue https://github.com/mozilla-mobile/fenix/issues/26511 can be observed at launch;
- Accessing Settings and returning to homepage, removes the wallpaper from homepage, but after opening a page or restarting the app, the wallpaper is displayed again.

### Device name

Google Pixel 6 

### Android version

Android 13

### Firefox release type

Firefox Nightly

### Firefox version

Nightly 106.0a1 from 09/14 and 09/15

### Additional information

https://user-images.githubusercontent.com/89388888/190348404-c586b6ad-eb46-4fd9-b9af-924690077303.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21576)

**Comment by quadriflax** [comment]
Starting with the below version, I'm getting a pop up message that reads "Couldn't change wallpaper" on my Samsung Tab A 10.1 (2019) with A11. This happens when loading to homescreen. I suspect it's related to this and the previously closed out bugs causing it to not load/become unresponsive, so I'm making note here.

106.0a1 (Build #2015904169), 7fc96b01b+
AC: 106.0.20220914204738, 592037c406
GV: 106.0a1-20220914154803
AS: 94.2.0

**Comment by delia-pop** [comment]
Hi, @quadriflax, thank for the input. Indeed, the "Couldn't change wallpaper" snackbar message should be displayed when the wallpaper couldn't be selected due to a certain error. I did not encounter this so far but I will keep an eye on it.  Can you share a video recording of the issue?

**Comment by quadriflax** [comment]
> Hi, @quadriflax, thank you for the input. Indeed, the "Couldn't change wallpaper" snackbar message should be displayed when the wallpaper couldn't be selected due to a certain error. I did not encounter this so far but I will keep an eye on it. Can you share a video recording of the issue?

I can do a screenshot. This just pops up and goes away a second or two later.

![Screenshot_20220915-053921_Firefox Nightly](https://user-images.githubusercontent.com/46144580/190460117-8cb4140e-1dea-44d4-bd2a-def04f8d942c.jpg)

**Comment by delia-pop** [comment]
Verified as fixed on the latest Nightly 106 from 09/16 with Google Pixel 6 (Android 13). The wallpaper is no longer removed after returning from Settings.
We tested this for an upgrading user with a wallpaper set on an older Fenix version, since due to https://github.com/mozilla-mobile/fenix/issues/26997 wallpapers are not displayed. 

https://user-images.githubusercontent.com/89388888/190572667-c4360a73-d071-414e-90f3-bd965ccf70a8.mp4
